### PR TITLE
Add Emiller Personal planck layout

### PR DESCRIPTION
**Pull request by Emiller88**
Featuring a vim movement layer

### CODE REVIEW

File: keyboards/planck/keymaps/emiller/config.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/planck/keymaps/emiller/config.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#define CONFIG_USER_H
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/config.h]:
```suggestion
```

File: keyboards/planck/keymaps/emiller/config.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/config.h]: This line is not actually needed, and should be removed. 

```suggestion
```

Specifically, the build script automatically includes all of the config.h files, so this is not necessary.

File: keyboards/planck/keymaps/emiller/config.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
+#endif
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/config.h]:
```suggestion
```

File: keyboards/planck/keymaps/emiller/keymap.c [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "planck.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/keymap.c]: Could you replace this:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/planck/keymaps/emiller/keymap.c [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/planck/keymaps/emiller/keymap.c [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Ctrl | GUI  | Alt  | VIM  | Lower|    Space    |Raise | NUM  | Vol- | Vol+ | Del  |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = {
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/keymap.c]: Could you convert this to the format found in the default planck keymap? 

https://github.com/qmk/qmk_firmware/blob/1df10ae3bfc146805c75b06dc0c126cd1ef3a60a/keyboards/planck/keymaps/default/keymap.c#L57-L62

File: keyboards/planck/keymaps/emiller/keymap.c [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+          PLAY_SONG(plover_gb_song);
+        #endif
+        layer_off(_PLOVER);
+      }
+      return false;
+      break;
+    case VIM:
+      if (record->event.pressed) {
+        layer_on(_VIM);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/keymap.c]: These shouldn't be needed, and you should be able to just use `MO(_VIM)` in your keymap.

File: keyboards/planck/keymaps/emiller/rules.mk [hunk 9]
```diff
@@ -0,0 +1,5 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/rules.mk]: Could you remove this (ifndef/include/endif) block? 

It's not needed, and is a holdover from the old build system.

**Comment by drashna** [comment]
@Emiller88 I've flagged a number of changes.  Could you make these changes? 

Most of them, you should be able to "apply" from github interface.

File: keyboards/planck/keymaps/emiller/config.h [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
```
Comment (drashna) [inline keyboards/planck/keymaps/emiller/config.h]: This needs to be removed, too.

```suggestion
```

**Comment by drashna** [comment]
Any update on this?

**Comment by Emiller88** [comment]
I will try to get to it this weekend.

**Comment by Emiller88** [comment]
@drashna Let me know if there's anything else to fix!

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**